packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/ViewAQLInterpreterFactory.java

```diff
@@ -52,7 +53,8 @@ public ViewAQLInterpreterFactory(List<IJavaServiceProvider> javaServiceProviders
         this.javaServiceProviders = new ArrayList<>();
         this.javaServiceProviders.addAll(Objects.requireNonNull(javaServiceProviders));
         IServiceProvider nodeServiceProvider = (IReadOnlyQueryEnvironment queryEnvironment, boolean forWorkspace) -> ServiceUtils.getReceiverServices(null, Node.class).stream().toList();
-        this.javaServiceProviders.add((View view) -> List.of(CanonicalServices.class, DiagramServices.class, nodeServiceProvider.getClass()));
+        IServiceProvider treeItemServiceProvider = (IReadOnlyQueryEnvironment queryEnvironment, boolean forWorkspace) -> ServiceUtils.getReceiverServices(null, TreeItem.class).stream().toList();
+        this.javaServiceProviders.add((View view) -> List.of(CanonicalServices.class, DiagramServices.class, nodeServiceProvider.getClass(), treeItemServiceProvider.getClass()));
```

Comment:
Why this dependency between the interpreter and the tree dsl?

Reply:
To let specifier define preconditions on their context menu entries that access the selected item. For example, the `ExpandAll` entry precondition is `aql:selectedTreeItem.isHasChildren()`, which is more convenient than creating a service to retrieve this value. Let me know if you don't want such dependency and I'll create a service for that.